manifests/resource/map.pp

```diff
@@ -63,22 +63,14 @@
 
 
 define nginx::resource::map (
-  $string,
-  $mappings,
-  $default    = undef,
-  $ensure     = 'present',
-  $hostnames  = false
+  String $string,
+  Variant[Array, Hash] $mappings,
+  Optional[String] $default         = undef,
+  Enum['absent', 'present'] $ensure = 'present',
+  Boolean $hostnames                = false
 ) {
-  validate_string($string)
   validate_re($string, '^.{2,}$',
     "Invalid string value [${string}]. Expected a minimum of 2 characters.")
-  if ! ( is_array($mappings) or is_hash($mappings) ) {
```

Comment:
as discussed on IRC: we can kep the fail() call but this will never be reached. I wanted to move it as comments to the header documentation, but I jut noticed that it is already present.